lib/livebook/runtime.ex

```diff
@@ -533,12 +536,12 @@ defprotocol Livebook.Runtime do
   @doc """
   Sets the given environment variables.
   """
-  @spec put_system_envs(t(), list({String.t(), String.t()})) :: :ok
+  @spec put_system_envs(t(), list(Secret.t() | EnvVar.t())) :: :ok
   def put_system_envs(runtime, envs)
 
   @doc """
   Unsets the given environment variables.
   """
-  @spec delete_system_envs(t(), list(String.t())) :: :ok
-  def delete_system_envs(runtime, names)
+  @spec delete_system_envs(t(), list(Secret.t() | EnvVar.t())) :: :ok
+  def delete_system_envs(runtime, envs)
```

Comment:
I think it's actually better to have the string versions to keep the `Runtime` protocol simpler. In particular, note that the runtime node does not have the `Secret`/`EnvVar` modules, and while it's not an issue for map pattern matching, I'd keep the API similar to `System.put_env/delete_env`.

Reply:
Agreed. The runtime does not need to care about Secret/EnvVar.